subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/DownloadableGradleDistribution.groovy

```diff
@@ -16,68 +16,48 @@
 
 package org.gradle.integtests.fixtures.executer
 
-import org.gradle.api.Action
-import org.gradle.cache.CacheBuilder
-import org.gradle.cache.FileLockManager
-import org.gradle.cache.PersistentCache
-import org.gradle.cache.internal.CacheFactory
-import org.gradle.cache.internal.DefaultCacheFactory
-import org.gradle.cache.internal.DefaultFileLockManager
-import org.gradle.cache.internal.DefaultProcessMetaDataProvider
-import org.gradle.cache.internal.locklistener.NoOpFileLockContentionHandler
-import org.gradle.internal.concurrent.DefaultExecutorFactory
-import org.gradle.internal.progress.NoOpProgressLoggerFactory
+
 import org.gradle.test.fixtures.file.TestFile
-import org.gradle.testfixtures.internal.NativeServicesTestFixture
 import org.gradle.util.GradleVersion
 
-import static org.gradle.cache.internal.filelock.LockOptionsBuilder.mode
-
 abstract class DownloadableGradleDistribution extends DefaultGradleDistribution {
-
-    private static final CACHE_FACTORY = createCacheFactory()
-
-    private static CacheFactory createCacheFactory() {
-        return new DefaultCacheFactory(
-                new DefaultFileLockManager(
-                        new DefaultProcessMetaDataProvider(
-                                NativeServicesTestFixture.getInstance().get(org.gradle.internal.nativeintegration.ProcessEnvironment)),
-                        20 * 60 * 1000 // allow up to 20 minutes to download a distribution
-                        , new NoOpFileLockContentionHandler()), new DefaultExecutorFactory(), new NoOpProgressLoggerFactory())
-    }
-
-    protected TestFile versionDir
-    private PersistentCache cache
+    private TestFile versionDir
+    private final ExclusiveFileAccessManager fileAccessManager = new ExclusiveFileAccessManager(120000, 200)
 
     DownloadableGradleDistribution(String version, TestFile versionDir) {
         super(GradleVersion.version(version), versionDir.file("gradle-$version"), versionDir.file("gradle-$version-bin.zip"))
         this.versionDir = versionDir
     }
 
     TestFile getBinDistribution() {
-        download()
+        downloadIfNecessary()
         super.getBinDistribution()
     }
 
     TestFile getGradleHomeDir() {
-        download()
+        downloadIfNecessary()
         super.getGradleHomeDir()
     }
 
-    private void download() {
-        if (cache == null) {
-            def downloadAction = { cache ->
+    private downloadIfNecessary() {
+        download(super.getBinDistribution(), super.getGradleHomeDir(), versionDir)
+    }
+
+    private void download(TestFile distributionZip, TestFile gradleHomeDir, TestFile versionDir) {
```

Comment:
💅 This method should also be named `downloadIfNeccessary`.

Reply:
done